packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart

```diff
@@ -11,42 +11,51 @@ import '../../../shared/primitives/class_name.dart';
 import '../../../shared/primitives/instance_set_view.dart';
 
 class HeapClassSampler extends ClassSampler {
-  HeapClassSampler(this.className, this.objects);
+  HeapClassSampler(this.objects, this.heap);
 
-  final HeapClassName className;
   final SingleClassStats objects;
+  final AdaptedHeapData heap;
 
   IsolateRef get _mainIsolateRef =>
       serviceManager.isolateManager.mainIsolate.value!;
 
-  Future<InstanceRef> _oneInstance() async {
+  Future<InstanceRef?> _oneInstance() async {
     final isolateId = _mainIsolateRef.id!;
 
     // TODO(polina-c): It would be great to find out how to avoid full scan of classes.
     final theClass = (await serviceManager.service!.getClassList(isolateId))
         .classes!
-        .firstWhere((ref) => className.matches(ref));
+        .firstWhere((ref) => objects.heapClass.matches(ref));
 
     final instances = await serviceManager.service!.getInstances(
       isolateId,
       theClass.id!,
       1,
     );
 
-    return instances.instances!.first as InstanceRef;
+    final result = instances.instances!.first;
+
+    if (result is InstanceRef) return result;
```

Comment:
In what scenario would this not be an `InstanceRef`? If it's only an error case, consider throwing an exception rather than having a nullable return type.